### PR TITLE
nonce for devtools style

### LINKED ISSUE #3495: CSP enabled DevTool show errors (insert inline style)
### Describe the bug

When CSP enabled, (Style-SRC='xxx')

the console error
```
 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'nonce-0mihlgp7' 'report-sample'
```

### Your minimal, reproducible example

https://codesandbox.io/s/heuristic-cannon-it3v0r?file=/public/index.html

### Steps to reproduce

1. start the app
2. open the app in single window
3.  open the console and refresh

### Expected behavior

No errors or passing the nonce as params

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

Browsers

### react-query version

3.34.19

### TypeScript version

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by NewFuture**
* pass nonce to devtools
* __webpack__nonce__
* set style scoped 

fix #3495

### CODE REVIEW

File: src/devtools/devtools.tsx [hunk 1]
```diff
@@ -229,6 +238,7 @@ export function ReactQueryDevtools({
       <ThemeProvider theme={theme}>
         <ReactQueryDevtoolsPanel
           ref={panelRef as any}
+          styleNonce={styleNonce || (window as any).__webpack_nonce__}
```
Comment (TkDodo) [inline src/devtools/devtools.tsx:241]: I'm fine with the style nonce, but I'd rather not couple this to window and `__webpack_nonce__`. It should be easily doable in user land to forward this:

```
<ReactQueryDevtools styleNonce={window.__webpack_nonce__}
```

File: src/devtools/devtools.tsx [hunk 2]
```diff
@@ -467,6 +483,9 @@ export const ReactQueryDevtoolsPanel = React.forwardRef<
         {...panelProps}
       >
         <style
+          scoped
+          nonce={styleNonce}
+          data-testid="react-query-devtools-panel-style"
```
Comment (TkDodo) [inline src/devtools/devtools.tsx:488]: we can just find the `style` tag without needing a test-id I guess, just by doing:

```
const { container } = render(<Page />)
const button = container.querySelector('style')
```

**Review by TkDodo — CHANGES_REQUESTED** [review]

File: src/devtools/devtools.tsx (hunk 1 above)
Comment (NewFuture) [reply src/devtools/devtools.tsx]: removed

**Comment by NewFuture** [comment]
@TkDodo help review 😸

File: src/devtools/devtools.tsx [hunk 3]
```diff
@@ -467,6 +483,8 @@ export const ReactQueryDevtoolsPanel = React.forwardRef<
         {...panelProps}
       >
         <style
+          scoped
```
Comment (TkDodo) [inline src/devtools/devtools.tsx:486]: what does this attribute do please? according to the docs, it's a non-standard, deprecated thing?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLStyleElement/scoped

Comment (NewFuture) [reply src/devtools/devtools.tsx:486]: I did not realize that it's deprecated now.
The purpose is to make this style with no side effects on the whole document.

Comment (TkDodo) [reply src/devtools/devtools.tsx:486]: I still think we should remove it please.

Comment (NewFuture) [reply src/devtools/devtools.tsx]: yeah, removed

File: src/devtools/tests/devtools.test.tsx [hunk 4]
```diff
@@ -391,4 +391,21 @@ describe('ReactQueryDevtools', () => {
     expect(queries[1]?.textContent).toEqual(query2Hash)
     expect(queries[2]?.textContent).toEqual(query3Hash)
   })
+
+  it('style should have scoped and nonce', async () => {
```
Comment (TkDodo) [inline src/devtools/tests/devtools.test.tsx:395]:
```suggestion
  it('style should have a nonce', async () => {
```

**Review by TkDodo — APPROVED** [review]

**Pull request merged**